test/e2e/network/dual_stack.go

```diff
@@ -0,0 +1,308 @@
+												{
+													Key:      "test",
+													Operator: metav1.LabelSelectorOpIn,
+													Values:   []string{"dualstack-test-server"},
+												},
+											},
+										},
+										TopologyKey: "kubernetes.io/hostname",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		clientDeploymentSpec := &apps.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   networkUtilsDeploymentName,
+				Labels: map[string]string{"test": "dualstack-test-client"},
+			},
+			Spec: apps.DeploymentSpec{
+				Replicas: &replicas,
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"test": "dualstack-test-client"}},
+				Strategy: apps.DeploymentStrategy{
+					Type: apps.RollingUpdateDeploymentStrategyType,
+				},
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"test": "dualstack-test-client"},
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:    "dualstack-test-client",
+								Image:   networkUtilImage,
+								Command: []string{"/bin/sh", "-c", "--"},
+								Args:    []string{"while true; do sleep 30; done;"},
+							},
+						},
+						Affinity: &v1.Affinity{
+							PodAntiAffinity: &v1.PodAntiAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+									{
+										LabelSelector: &metav1.LabelSelector{
+											MatchExpressions: []metav1.LabelSelectorRequirement{
+												{
+													Key:      "test",
+													Operator: metav1.LabelSelectorOpIn,
+													Values:   []string{"dualstack-test-client"},
+												},
+											},
+										},
+										TopologyKey: "kubernetes.io/hostname",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		serverDeployment, err := cs.AppsV1().Deployments(f.Namespace.Name).Create(serverDeploymentSpec)
+		framework.ExpectNoError(err)
+
+		clientDeployment, err := cs.AppsV1().Deployments(f.Namespace.Name).Create(clientDeploymentSpec)
+		framework.ExpectNoError(err)
+
+		err = e2edeploy.WaitForDeploymentComplete(cs, serverDeployment)
+		framework.ExpectNoError(err)
+		err = e2edeploy.WaitForDeploymentComplete(cs, clientDeployment)
+		framework.ExpectNoError(err)
+
+		serverPods, err := e2edeploy.GetPodsForDeployment(cs, serverDeployment)
+		framework.ExpectNoError(err)
+
+		clientPods, err := e2edeploy.GetPodsForDeployment(cs, clientDeployment)
+		framework.ExpectNoError(err)
+
+		assertNetworkConnectivity(f, *serverPods, *clientPods)
+	})
+})
+
+var (
+	duration     = "10s"
+	pollInterval = "1s"
+	timeout      = 10
+)
+
+func assertNetworkConnectivity(f *framework.Framework, serverPods v1.PodList, clientPods v1.PodList) {
+	// curl from each client pod to all server pods to assert connectivity
+	var serverIPs []string
+	for _, pod := range serverPods.Items {
+		if pod.Status.PodIPs == nil || len(pod.Status.PodIPs) != 2 || !isIPv6(pod.Status.PodIPs[1].IP) {
```

Comment:
this means that the second podIP it's always IPv6 and I guess that is not correct, we should check that both PodIps belong only to different families